javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue1668.java

```diff
@@ -0,0 +1,62 @@
+package com.github.javaparser.symbolsolver;
+
+import com.github.javaparser.*;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.body.VariableDeclarator;
+import com.github.javaparser.ast.expr.VariableDeclarationExpr;
+import com.github.javaparser.resolution.declarations.ResolvedFieldDeclaration;
+import com.github.javaparser.resolution.types.ResolvedType;
+import com.github.javaparser.symbolsolver.model.resolution.TypeSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class Issue1668 {
+
+    private JavaParser javaParser;
+
+    @Before
+    public void setUp() {
+        TypeSolver typeSolver = new ReflectionTypeSolver();
+        ParserConfiguration config = new ParserConfiguration();
+        config.setSymbolResolver(new JavaSymbolSolver(typeSolver));
+        javaParser = new JavaParser(config);
+    }
+
+    @Test
+    public void test() {
+        String code = String.join(System.lineSeparator(),
+                "public class X {",
+                "   public static void main(String[] args) {",
+                "       String s = \"a,b,c,d,e\";",
+                "       String[] stringArray = s.split(',');",
+                "   }",
+                "}"
+        );
+
+        ParseResult<CompilationUnit> parseResult = javaParser.parse(ParseStart.COMPILATION_UNIT, Providers.provider(code));
+        assertTrue(parseResult.isSuccessful());
+        assertTrue(parseResult.getResult().isPresent());
+
+        CompilationUnit compilationUnit = parseResult.getResult().get();
+        Optional<VariableDeclarator> stringArray = compilationUnit.findFirst(VariableDeclarator.class, variableDeclarator ->
+                variableDeclarator.getNameAsString()
+                        .equals("stringArray"));
+        assertTrue(stringArray.isPresent());
+        VariableDeclarator variableDeclarator = stringArray.get();
+        variableDeclarator.getParentNode().ifPresent(parent -> {
```

Comment:
Don't we expect it to be present? Should we assert it is present?

Reply:
It should indeed be present, imo the rest of the test shouldn't even continue if for some case the VariableDeclarator has not been found. But on the other hand this is not what the test is for, so this assert can be dropped